toolset/utils/docker_helper.py

```diff
@@ -10,98 +9,104 @@
 from colorama import Fore, Style
 
 from toolset.utils.output_helper import log
-from toolset.utils.metadata_helper import gather_tests
 from toolset.utils.database_helper import test_database
 
 
-def clean(benchmarker_config):
-    '''
-    Cleans all the docker images from the system
-    '''
-    # Clean the app server images
-    client = docker.DockerClient(
-        base_url=benchmarker_config.server_docker_host)
-
-    client.images.prune()
-    for image in client.images.list():
-        if len(image.tags) > 0:
-            # 'techempower/tfb.test.gemini:0.1' -> 'techempower/tfb.test.gemini'
-            image_tag = image.tags[0].split(':')[0]
-            if image_tag != 'techempower/tfb' and 'techempower' in image_tag:
-                client.images.remove(image.id, force=True)
-    client.images.prune()
-
-    # Clean the database server images
-    client = docker.DockerClient(
-        base_url=benchmarker_config.database_docker_host)
-
-    client.images.prune()
-    for image in client.images.list():
-        if len(image.tags) > 0:
-            # 'techempower/tfb.test.gemini:0.1' -> 'techempower/tfb.test.gemini'
-            image_tag = image.tags[0].split(':')[0]
-            if image_tag != 'techempower/tfb':
-                client.images.remove(image.id, force=True)
-    client.images.prune()
-
-
-def __build(base_url, path, build_log_file, log_prefix, dockerfile, tag):
-    '''
-    Builds docker containers using docker-py low-level api
-    '''
-
-    with open(build_log_file, 'w') as build_log:
-        try:
-            client = docker.APIClient(base_url=base_url)
-            output = client.build(
-                path=path,
-                dockerfile=dockerfile,
-                tag=tag,
-                forcerm=True,
-                pull=True)
-            buffer = ""
-            for token in output:
-                if token.startswith('{"stream":'):
-                    token = json.loads(token)
-                    token = token[token.keys()[0]].encode('utf-8')
-                    buffer += token
-                elif token.startswith('{"errorDetail":'):
-                    token = json.loads(token)
-                    raise Exception(token['errorDetail']['message'])
-                while "\n" in buffer:
-                    index = buffer.index("\n")
-                    line = buffer[:index]
-                    buffer = buffer[index + 1:]
-                    log(line,
+class DockerHelper:
+
+    def __init__(self, benchmarker = None):
+        self.benchmarker = benchmarker
+
+        self.client = docker.DockerClient(
+            base_url=self.benchmarker.config.client_docker_host)
+        self.server = docker.DockerClient(
+            base_url=self.benchmarker.config.server_docker_host)
+        self.database = docker.DockerClient(
+            base_url=self.benchmarker.config.database_docker_host)
```

Comment:
I have a **slight** concern here that the client may close a connection if idle for too long. May not come up in practice, but maybe dig into the API to see if there is a timeout (or something similar).

Reply:
I'm not sure this actually opens the connection until you start sending requests to the daemon